### PR TITLE
[Quest API] Add $client->SetGMStatus()

### PR DESCRIPTION
**Pull request by splose**
A Server Operator requested a way to change account status through a perl script.

### CODE REVIEW

**Comment by fryguy503** [comment]
All quest functions need to include both LUA and Perl. Please reference previously implemented PR's for examples on adding LUA functionality.

**Comment by splose** [comment]
I'm not adding it to lua.

**Comment by SecretsOTheP** [comment]
@mackal - review it, see if you can merge it in and I'll open up an issue ticket for lua support for this function in the future.

@TurmoilToad, @fryguy503
The context is this was asked to be added by a community member named kimura, who uses exclusively Perl.
I understand you guys want it ported to lua (and frankly, it takes 30 seconds to do so, but that's not the point).

Splose is **trying to help community members** and this is functionality that extends beyond Kimura's request. I get why you want it on lua. Furthermore he is trying to contribute in a positive manner. I would like to see that continue and for trust (no pun intended) to be improved by already strained relationships.

I don't think it's acceptable to keep denying pull requests like this when the code works. If you, as a repository maintainer, have had interpersonal issues with a contributor in the past, and put them under a myopic lens to 'judge' their character, **you have already failed as you are putting yourself ahead of contributors**. I know that's exactly what is going on here and I am sick of staying silent.
 
@splose has done a majority of the work for this functionality while working on his own server. He went out of his way for someone else, submitted a pull request, and more importantly is putting focus on contributions. Asking a growing contributor to spend more time than they have, is akin to asking me to port any functionality over from classless3.0 when I work a full time job on top of my hobby. It's actually the sole reason why I have no time to port over classless3's big bags despite having a clear plan in my mind to do so (ie; manually run 65535 queries to reassign 100 slots)

Quite frankly, asking him to add lua when anyone else in this community is capable of doing it themselves is a bit unreasonable request of a server operator. A server is a full time job, even in preproduction. The fact that Splose went out of his way to make a contribution should speak volumes to his commitment to making the emulator better.

@Akkadius - I am sick of the corporate BS that you and the newly appointed schutzstaffel seem to be peddling out of EQEmulator and if you wish to continue this discussion in private, contact my cell to apologize to me and clear the air.

**Comment by SecretsOTheP** [comment]
@Akkadius  - Side note, I have time to do this right now, and I would've just made an issue ticket, merged it in and did the functionality myself in 5 minutes and you would've had all the fucking accountability you would have wanted. But now work starts in 5 minutes, I have 8 hours to focus solely on that, and 

But apparently I don't have write access to the repository I spent years contributing to, and the fourth reich already drove me out of Discord with being confronted for having a wild opinion. Work starts in 5 minutes, I have 8 hours to focus solely on that, and you'll just have to wait.

Side note, the code of conduct is fine. This is the exact issue with fucking Marxism that most saw; it's not that the idea sucks, it's that the execution is poor and the wrong people fuck it up. Don't be the CCP.

**Comment by fryguy503** [comment]
No denying of any PR was done here and that is why it is still open and pending review. Please keep this to comments related to this specific PR and its code implications.

**Comment by SecretsOTheP** [comment]
> 
> 
> No denying of any PR was done here and that is why it is still open and pending review. Please keep this to comments related to this specific PR and its code implications.

I already told you what we will do. See below:

#@mackal - review it, see if you can merge it in and I'll open up an issue ticket for lua support for this function in the future.

I will personally handle implementing it.

You chose to ignore my conversation as it included subjects you don't like.
Please keep in mind the code of conduct when interacting with other people on the project, especially senior contributors like myself.

File: zone/perl_client.cpp [hunk 1]
```diff
@@ -3413,6 +3413,22 @@ XS(XS_Client_ReadBook) {
...
 }
 
+XS(XS_Client_SetGMStatus); /* prototype to pass -Wmissing-prototypes */
+XS(XS_Client_SetGMStatus) {
+	dXSARGS;
+	if (items != 2)
+		Perl_croak(aTHX_ "Usage: Client::SetGMStatus(THIS, int newStatus)"); // @categories Script Utility
+	{
+		Client *THIS;
+		uint32 accID = THIS->AccountID();
```
Comment (mackal) [inline zone/perl_client.cpp:3423]: This line appears to be left over. Can be removed.

**Review by mackal — APPROVED** [review]

**Comment by mackal** [comment]
I'll merge this since Secrets said they'll do the lua part.

**Pull request merged**